nix/packages/stub.nix

```diff
@@ -0,0 +1,60 @@
+{ rust, rustPlatform, clippy, rustfmt, stdenv, lib, runCommand, enableFmt ? false, enableLint ? false }:
+
+let
+  targetSpec = rust.toRustTargetSpec stdenv.hostPlatform;
+  targetIsJSON = lib.hasSuffix ".json" targetSpec;
+  shortTarget =
+    if targetIsJSON then
+      (lib.removeSuffix ".json" (builtins.baseNameOf "${targetSpec}"))
+    else targetSpec;
+in
+rustPlatform.buildRustPackage
+  ({
+    pname = "lanzastub";
+    version = "0.3.0";
+    src = runCommand "src" { } ''
+      install -D ${../../rust/stub/Cargo.toml} $out/Cargo.toml
+      install -D ${../../rust/stub/Cargo.lock} $out/Cargo.lock
+      cp -r ${../../rust/stub/src} $out/src
+    '';
+
+    nativeBuildInputs = lib.optional enableLint clippy ++ lib.optional enableFmt rustfmt;
+
+    cargoLock = {
+      lockFile = ../../rust/stub/Cargo.lock;
+    };
+
+    # Necessary because our `cc-wrapper` doesn't understand MSVC link options.
+    RUSTFLAGS = "-Clinker=${stdenv.cc.bintools}/bin/${stdenv.cc.targetPrefix}ld.lld -Clinker-flavor=lld-link";
+    # Necessary because otherwise we will get (useless) hardening options in front of
+    # -flavor link which will break the whole command-line processing for the ld.lld linker.
+    hardeningDisable = [ "all" ];
+
+    meta = with lib; {
+      description = "Lanzaboote UEFI stub for SecureBoot enablement on NixOS systems";
+      homepage = "https://github.com/nix-community/lanzaboote";
+      license = licenses.mit;
+      platforms = [ "x86_64-windows" "aarch64-windows" "i686-windows" ];
```

Comment:
Did we actually get building for aarch64 to work?

Reply:
`error: Package ‘compiler-rt-14.0.6’ in /nix/store/wr7rzhqpb71986npaa2fvbjg302yb3r7-source/pkgs/development/compilers/llvm/14/compiler-rt/default.nix:128 is not available on the requested hostPlatform:`

but passing the classical "unsupported systems ok" flag yields you
```
❯ llvm-objdump --section-headers /nix/store/xfjgpsn98yw3i0bc071zdk4606ij10bg-lanzastub-aarch64-windows-0.3.0/bin/lanzaboote_stub.efi 

/nix/store/xfjgpsn98yw3i0bc071zdk4606ij10bg-lanzastub-aarch64-windows-0.3.0/bin/lanzaboote_stub.efi:	file format coff-arm64

Sections:
Idx Name          Size     VMA              Type
  0 .text         00014770 0000000140001000 TEXT
  1 .rdata        000047f0 0000000140016000 DATA
  2 .data         00000070 000000014001b000 DATA
  3 .reloc        000003a8 000000014001c000 DATA
  ```

on a native aarch64 machine

Reply:
Yes, it wasn't a question whether it would build, but whether it would work.

Remember the discussion about how aarch64 UEFI is not the same as aarch64 Windows that caused the argument over triples in the first place.

Reply:
Unfortunately, testing until the end requires some support in systemd which is only on the main branch at the moment.